hera_mc/cm_hookup.py

```diff
@@ -173,46 +64,128 @@ def get_hookup(self, hpn, rev, port, at_date, state_args, exact_match=False):
-                                       parts[hpnr]['part'].start_date,
-                                       parts[hpnr]['part'].stop_date):
-                continue
-            if len(parts[hpnr]['connections']['ordered-pairs'][0]) == 0:
-                continue
-            how_many_to_do = self.__wade_through_the_messy_stuff(parts[hpnr], True)
-
-            parts[hpnr]['input_ports'].sort()
-            parts[hpnr]['output_ports'].sort()
-            if type(port) == str and port.lower() == 'all':
-                port_query_loop = parts[hpnr]['input_ports']
-                if len(port_query_loop) == 0:
-                    port_query_loop = parts[hpnr]['output_ports']
-            else:  # This to handle range of port_query possibilities outside of 'all'
-                print("Not really supported.  Need a way to check if upstream or "
-                      "downstream port etcetc")
-                if type(port) != list:
-                    port_query_loop = [port]
+        pols_to_do = self.__get_pols_to_do(hpn, port, check_pol=False)
 
-            for i, p in enumerate(port_query_loop):
-                for sn in range(how_many_to_do):
-                    hukey = _make_hookup_key(parts[hpnr]['part'].hpn,
-                                             parts[hpnr]['part'].hpn_rev, p, str(sn))
-                    hookup_dict['hookup'][hukey] = self.__follow_hookup_stream(
-                        parts[hpnr]['part'].hpn, parts[hpnr]['part'].hpn_rev, p, sn)
-                    if len(hookup_dict['hookup'][hukey]) > col_len_max[0]:
-                        col_len_max[0] = len(hookup_dict['hookup'][hukey])
-                        col_len_max[1] = hukey
-        if len(hookup_dict['hookup'].keys()) == 0:
-            print(hpn, rev, 'not active')
-            return None
+        for k, part in parts.iteritems():
+            if not cm_utils._is_active(self.at_date, part['part'].start_date, part['part'].stop_date):
+                continue
+            #if len(part['connections']['ordered-pairs'][0]) == 0:
+            #    continue
+            hookup_dict['hookup'][k] = {}
+            for pol in pols_to_do:
+                hookup_dict['hookup'][k][pol] = self.__follow_hookup_stream(part['part'].hpn, part['part'].hpn_rev, pol)
+                if len(hookup_dict['hookup'][k][pol]) > col_len_max[0]:
+                    col_len_max[0] = len(hookup_dict['hookup'][k][pol])
+                    col_len_max[1] = k
+        hookup_dict = self.__add_hookup_timing(hookup_dict)
 
         hookup_dict['columns'] = self.__get_column_header(
-            hookup_dict['hookup'][col_len_max[1]])
+            hookup_dict['hookup'][col_len_max[1]][pol])
         if state_args['show_levels']:
             hookup_dict = self.__hookup_add_correlator_levels(
                 hookup_dict, state_args['levels_testing'])
+        return hookup_dict
+
+    def __get_pols_to_do(self, part, port, check_pol=False):
+        all_pols = ['e', 'n']
```

Comment:
My concern is that I'm not at all sure it's getting handled the same in the data processing codes and I know that the hera_qm tables are using 'x' and 'y', in large part because that's what the files use (and that's partly because of the limitations of the file formats). I just really want to make sure everyone is on the same page here and to decide where we're using which nomenclature.